test/Constraints/invalid_implicit_conversions.swift

```diff
@@ -23,6 +24,7 @@ func test(
   takesAutoclosure(mutRawPtr, &a) // expected-error {{cannot perform pointer conversion of value of type '[Int]' to autoclosure result type 'UnsafeMutableRawPointer'}}
   takesAutoclosure(mutPtr, &a) // expected-error {{cannot perform pointer conversion of value of type '[Int]' to autoclosure result type 'UnsafeMutablePointer<Int>'}}
   takesAutoclosure(ptr, &a) // expected-error {{cannot perform pointer conversion of value of type '[Int]' to autoclosure result type 'UnsafePointer<Int>'}}
+  takesAutoclosure(optPtr, &i) // expected-error {{cannot perform pointer conversion of value of type 'Int' to autoclosure result type 'UnsafePointer<Int>?'}}
```

Comment:
This is effectively a source break even though it's in an invalid code which should have been rejected, I think we might want to document this in `CHANGELOG` or similar document...

Reply:
This particular example doesn't compile currently with the error `'&' can only appear immediately in a call argument list`. But this change is still a source break with the examples above

Reply:
I tried this simplified version:
```swift
func foo(_: @autoclosure () -> UnsafePointer<Int>?) {}
var i: Int = 0
foo(&i)
```

Reply:
Ok interesting, looks like it has been diagnosed in 4.2, 5.0 and 5.1 (one of the snapshots  I have) which means that it's only a master regression.